semanticdb/semanticdb3/javacp.md

```diff
@@ -0,0 +1,551 @@
+`TYPE_PARAMETER` and formal parameter with kind `PARAMETER`.
+Notes:
+* When compiled with the compiler option `-parameters`, the name of method
+  parameters matches their name written in source. Otherwise, parameters have
+  the name `paramN` where `N` is the index of that given parameter starting at
+  index 0.
+
+**Field declarations** [\[83\]][83] are represented by a single symbol with
+the `FIELD` kind.
+
+**Constructor declarations** [\[90\]][90] are represented by a single symbol with
+name `<init>` and the `CONSTRUCTOR` kind. Constructor formal parameters are represented
+the same way as method declaration formal parameters.
+
+<a name="java-accessibility"></a>
+#### Accessibility
+
+```protobuf
+message Accessibility {
+  enum Tag {
+    UNKNOWN_ACCESSIBILITY = 0;
+    PRIVATE = 1;
+    PRIVATE_THIS = 2;
+    PRIVATE_WITHIN = 3;
+    PROTECTED = 4;
+    PROTECTED_THIS = 5;
+    PROTECTED_WITHIN = 6;
+    PUBLIC = 7;
+  }
+  Tag tag = 1;
+  string symbol = 2;
+}
+```
+
+In Java, [Accessibility](#accessibility) represents access control [\[87\]][87] of names
+<table>
+  <tr>
+    <td><b>Accessibility</b></td>
+    <td><b>Code</b></td>
+    <td><b>Explanation</b></td>
+  </tr>
+  <tr>
+    <td><code>PRIVATE</code></td>
+    <td><code>private void m() {}</code></td>
+    <td>
+      Can be accessed only from within the directly enclosing class.
+      <a href="https://docs.oracle.com/javase/specs/jls/se8/html/jls-6.html#jls-6.6.1">[88]</a>.
+    </td>
+  </tr>
+  <tr>
+    <td><code>PRIVATE_WITHIN</code></td>
+    <td><code>package x; class A {}</code></td>
+    <td>
+      A class, interface, class member or constructor declared without an access
+      modifier is implicitly private within within the package in which is
+      declared.
+      <a href="https://docs.oracle.com/javase/specs/jls/se8/html/jls-6.html#jls-6.6.1">[88]</a>.
+    </td>
+  </tr>
+  <tr>
+    <td><code>PROTECTED</code></td>
+    <td><code>protected void m() {}</code></td>
+    <td>
+      A protected member of constructor of an object can be accessed from
+      within: 1) the enclosing class, 2) all classes that are responsible for
+      the implementation of that object.
+      <a href="https://docs.oracle.com/javase/specs/jls/se8/html/jls-6.html#jls-6.6.2">[89]</a>.
+    </td>
+  </tr>
+  <tr>
+    <td><code>PUBLIC</code></td>
+    <td><code>public void m() {}</code></td>
+    <td>
+      Can be accessed from from any code provided that the compilation unit
+      in which it is declared is observable. Packages are always implicitly
+      public. Members of interfaces lacking interface modifiers are
+      implicitly public. Other members are public only if explicitly declared
+      `public`.
+      <a href="https://docs.oracle.com/javase/specs/jls/se8/html/jls-6.html#jls-6.6.1">[88]</a>.
+    </td>
+  </tr>
+</table>
+
+Notes:
+* `PRIVATE_THIS`, `PROTECTED_THIS`, `PROTECTED_WITHIN` are not supported in the Java language.
```

Comment:
What about Java annotations? I think they deserve a separate section, including the fact that `strictfp` and friends are mapped onto annotations.

Reply:
I added a section on java annotations and removed the usage of ACC_DEPRECATED that wasn't doing what I thought it was doing. Only ACC_STRICTFP is supported, added a not JLS annotations are unsupported.